pkg/services/ngalert/api/testing.go

```diff
@@ -0,0 +1,109 @@
+package api
+
+import (
+	"context"
+	"testing"
+
+	apimodels "github.com/grafana/grafana/pkg/services/ngalert/api/tooling/definitions"
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+	"github.com/grafana/grafana/pkg/services/ngalert/notifier"
+	"github.com/grafana/grafana/pkg/services/ngalert/store"
+	"github.com/grafana/grafana/pkg/services/secrets"
+)
+
+type FakeAlertmanagerProvider struct {
+	alertmanagers map[int64]Alertmanager
+}
+
+func newAlertmanagerProvider(t *testing.T) *FakeAlertmanagerProvider {
+	t.Helper()
+
+	return &FakeAlertmanagerProvider{
+		alertmanagers: map[int64]Alertmanager{},
+	}
+}
+
+func (f FakeAlertmanagerProvider) Setup(orgID int64) {
+	f.alertmanagers[orgID] = &FakeAlertmanager{}
+}
+
+func (f FakeAlertmanagerProvider) AlertmanagerFor(orgID int64) (Alertmanager, error) {
+	if am, ok := f.alertmanagers[orgID]; ok {
+		return am, nil
+	}
+	return nil, notifier.ErrNoAlertmanagerForOrg
+}
+
+type FakeAlertingStore struct {
+	orgsWithConfig map[int64]bool
+}
+
+func newFakeAlertingStore(t *testing.T) FakeAlertingStore {
+	t.Helper()
+
+	return FakeAlertingStore{
+		orgsWithConfig: map[int64]bool{},
+	}
+}
+
+func (f FakeAlertingStore) Setup(orgID int64) {
+	f.orgsWithConfig[orgID] = true
+}
+
+func (f FakeAlertingStore) GetLatestAlertmanagerConfiguration(query *models.GetLatestAlertmanagerConfigurationQuery) error {
+	if _, ok := f.orgsWithConfig[query.OrgID]; ok {
+		return nil
+	}
+	return store.ErrNoAlertmanagerConfiguration
+}
+
+type FakeSecretService struct{}
+
+func (f FakeSecretService) Encrypt(ctx context.Context, payload []byte, opt secrets.EncryptionOptions) ([]byte, error) {
+	return payload, nil
+}
+
+func (f FakeSecretService) Decrypt(ctx context.Context, payload []byte) ([]byte, error) {
+	return payload, nil
+}
+
+type FakeAlertmanager struct{}
```

Comment:
This one was a large interface. Mocks rather than handwritten fakes would have simplified this file a lot. I stayed with handwritten fakes for consistency, all the existing code uses fakes, but it seems like mocks would be preferable as we add more unit tests.

Reply:
Not needed anymore.